### PR TITLE
Flask: GET /api/participants, /api/datasets, /api/analyses, /api/pipelines

### PR DESCRIPTION
**Pull request by Madelinehazel**
Kevin's new fix for serialization

Closes #18

### CODE REVIEW

File: flask/app/routes.py
Comment (delvinso) [inline flask/app/routes.py:250]: Might be just preference but collapse the outside brackets of the dictionaries and lists for readability since there's a lot going on here?

**Comment by delvinso** [comment]
Tested it out, looks good to me. 
```
# truncated to first participant
[
  {
    "affected": true, 
    "created": "Wed, 23 Sep 2020 18:40:48 GMT", 
    "created_by": 1, 
    "family_codename": "1000", 
    "notes": "wahahaha", 
    "participant_codename": "agent 10", 
    "participant_id": 1, 
    "participant_type": "Father", 
    "sex": "Male", 
    "solved": null, 
    "tissue_samples": [
      {
        "created": "Wed, 23 Sep 2020 18:40:48 GMT", 
        "created_by": 1, 
        "datasets": [
          {
            "batch_id": null, 
            "capture_kit": null, 
            "condition": "GermLine", 
            "created": "Mon, 03 Feb 2020 00:00:00 GMT", 
            "created_by": 1, 
            "dataset_id": 1, 
            "dataset_type": "WES", 
            "extraction_protocol": "Something", 
            "input_hpf_path": null, 
            "library_prep_date": null, 
            "library_prep_method": null, 
            "notes": null, 
            "read_length": null, 
            "read_type": null, 
            "sequencing_centre": "CHEO", 
            "sequencing_date": null, 
            "sequencing_id": null, 
            "tissue_sample_id": 1, 
            "updated": "Mon, 28 Sep 2020 14:23:31 GMT", 
            "updated_by": 1
          }
        ], 
        "extraction_date": "Sat, 04 Jan 2020 00:00:00 GMT", 
        "notes": null, 
        "participant_id": 1, 
        "tissue_processing": null, 
        "tissue_sample_id": 1, 
        "tissue_sample_type": "Blood", 
        "updated": "Wed, 23 Sep 2020 18:40:48 GMT", 
        "updated_by": 1
      }
    ], 
    "updated": "Mon, 28 Sep 2020 17:38:07 GMT", 
    "updated_by": 1
}
]
```

File: flask/app/models.py
Comment (kevinlul) [inline flask/app/models.py:80]: Extra space

File: flask/app/routes.py
Comment (kevinlul) [inline flask/app/routes.py:11]: Put dataclasses up top with the rest of the standard library imports

File: flask/app/routes.py
Comment (kevinlul) [reply flask/app/routes.py:250]: What do you think of this approach Delvin?

File: flask/app/routes.py
Comment (kevinlul) [reply flask/app/routes.py:237]: Does it not work on your end without the `endpoint` parameter?

**Review by kevinlul — CHANGES_REQUESTED** [review]

File: flask/app/routes.py
Comment (delvinso) [reply flask/app/routes.py:250]: I like it, it's a clean solution that doesn't require external packages or references to a million fields

File: flask/app/routes.py
Comment (delvinso) [reply flask/app/routes.py:237]: it works without the `endpoint` parameter for me

**Comment by kevinlul** [comment]
Will this be extended to the other endpoints in the old pull request #18 ?

**Comment by Madelinehazel** [comment]
> Will this be extended to the other endpoints in the old pull request #18 ?

yes! i can get started on those

**Comment by Madelinehazel** [comment]
@kevinlul @delvinso all GET endpoints from original PR are now included.

**Review by delvinso — APPROVED** [review]
good to go after Kevin's changes are merged

**Pull request merged**
